mm2src/coins/z_coin.rs

```diff
@@ -734,6 +740,47 @@ pub async fn z_coin_from_conf_and_params(
     builder.build().await
 }
 
+/// open file and calculate its sha256 digest as lowercase hex string
+fn sha256_digest(path: &PathBuf) -> Result<String, ZCoinBuildError> {
+    let input = File::open(path)?;
+    let mut reader = BufReader::new(input);
+
+    let digest = {
+        let mut hasher = Sha256::new();
+        let mut buffer = [0; 1024];
+        loop {
+            let count = reader.read(&mut buffer)?;
+            if count == 0 {
+                break;
+            }
+            hasher.update(&buffer[..count]);
+        }
+        format!("{:x}", hasher.finalize())
+    };
+    Ok(digest)
+}
```

Comment:
I think this looks like a  general purpose function that could be placed in common module for reusability.

Reply:
Done